### PR TITLE
AP_Common: fail nuttx build

### PR DESCRIPTION
**Pull request by rmackay9**
This is my guess as to how we can fail the NuttX builds ahead of removing NuttX from the build environment.  Perhaps there are other ways including failing during ./waf's configure step which would save some time for the developer because this compile time failure can take a couple of minutes to appear.

This causes an error to be printed on the console when the build fails, 

> NuttX builds have been deprecated in favour of ChibiOS.  Please use alternative build targets including fmuv2, fmuv3, fmuv4, fmuv5.

### CODE REVIEW

**Comment by tridge** [comment]
I think it is better to to it at the build system level, where we can present a reasonable bit of text explaining what the user can do. A #error just gives a big backtrace which is hard to parse by eye

**Comment by rmackay9** [comment]
I agree that it's not visible enough.  I'm a bit limited in my understanding of the build system but in any case, I'll close this item.
